kasa/iot/modules/dimmer.py

```diff
@@ -0,0 +1,279 @@
+                attribute_getter="fade_on_time",
+                attribute_setter="set_fade_on_time",
+                range_getter=lambda: (
+                    _td_to_ms(self.FADE_TIME_ABS_MIN),
+                    _td_to_ms(self.FADE_TIME_ABS_MAX),
+                ),
+                type=Feature.Type.Number,
+                category=Feature.Category.Config,
+            )
+        )
+
+        self._add_feature(
+            Feature(
+                device=self._device,
+                container=self,
+                id="dimmer_gentle_off_time",
+                name="Dimmer gentle off time",
+                icon="mdi:clock-in",
+                attribute_getter="gentle_off_time",
+                attribute_setter="set_gentle_off_time",
+                range_getter=lambda: (
+                    _td_to_ms(self.GENTLE_TIME_ABS_MIN),
+                    _td_to_ms(self.GENTLE_TIME_ABS_MAX),
+                ),
+                type=Feature.Type.Number,
+                category=Feature.Category.Config,
+            )
+        )
+
+        self._add_feature(
+            Feature(
+                device=self._device,
+                container=self,
+                id="dimmer_gentle_on_time",
+                name="Dimmer gentle on time",
+                icon="mdi:clock-out",
+                attribute_getter="gentle_on_time",
+                attribute_setter="set_gentle_on_time",
+                range_getter=lambda: (
+                    _td_to_ms(self.GENTLE_TIME_ABS_MIN),
+                    _td_to_ms(self.GENTLE_TIME_ABS_MAX),
+                ),
+                type=Feature.Type.Number,
+                category=Feature.Category.Config,
+            )
+        )
+
+        self._add_feature(
+            Feature(
+                device=self._device,
+                container=self,
+                id="dimmer_ramp_rate",
+                name="Dimmer ramp rate",
+                icon="mdi:clock-fast",
+                attribute_getter="ramp_rate",
+                attribute_setter="set_ramp_rate",
+                range_getter=lambda: (self.RAMP_RATE_ABS_MIN, self.RAMP_RATE_ABS_MAX),
+                type=Feature.Type.Number,
+                category=Feature.Category.Config,
+            )
+        )
+
+    def query(self) -> dict:
+        """Request Dimming configuration."""
+        req = merge(
+            self.query_for_command("get_dimmer_parameters"),
+            self.query_for_command("get_default_behavior"),
+        )
+
+        return req
+
+    @property
+    def config(self) -> dict[str, Any]:
+        """Return current configuration."""
+        return self.data["get_dimmer_parameters"]
+
+    @property
+    def threshold_min(self) -> int:
+        """Return the minimum dimming level for this dimmer."""
+        return self.config["minThreshold"]
+
+    async def set_threshold_min(self, min: int) -> dict:
+        """
+        Set the minimum dimming level for this dimmer.
```

Comment:
```suggestion
        """Set the minimum dimming level.
```
I'm not sure why this was not caught by our CI, but there should be [no newline before the docstring summary line](https://peps.python.org/pep-0257/#one-line-docstrings).

It's not that important as most tools probably account for that, but it's great to have for consistency all across the code base.

Reply:
It seemed like ruff was acting weird for me, perhaps I better do a clean pull of the repo, I had tried to keep the docstring on the first line, but ruff kept not liking it. I’ll go through and fix the docstrings.